File: C-CPP/WordGuess.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <ctype.h>
#include <iostream>
#include <limits>
#include <bits/stdc++.h>

int main(void) {
    char name[50];
    printf("What is your name? ");
    scanf("%s", name);
    printf("Hello %s, I want to play a game.\n", name);
    
    int i = 0;
    int turns = 10;
    int attempts = 0;
    char guess[50];
    char word[50];
    std::string guesses[turns];
    std::array<std::string, 10> words;
    while(true){
        printf("Enter a word: ");
        int result = scanf("%s", word);
        if(word[0] == 'q'){
            break;
        }
        words[i++] = word;
    }
    
    srand(time(NULL));
    int number = rand() % words.size();
    
    std::string wordToGuess = words.at(number);

    while(attempts < turns){
        printf("Guess a word: ");
        scanf("%s", guess);        
        guesses[attempts] = guess;
        attempts += 1;

        if(guess != word){
            printf("That is not correct!\n");
        }
        else{
            printf("That is correct!\n");
            printf("Found answer in attempt %d\n", attempts);
            break;
        }

        printf("Attemps made: %d\n", attempts);
    }

    printf("Guess history:\n");
    for(int i = 0; i < attempts; i++){
        printf("%s\n", guesses[i].c_str());
    }
}

File: NET/WordGuess.cs
using System;
using System.Collections.Generic;

namespace TSAF{
    public class WordGuess{
        public static void Main(){
            Console.Write("What is your name? ");
            string name = Console.ReadLine();
            Console.WriteLine("Hello " + name + ", I want to play a game.");

            List<string> guesses = new List<string>();
            List<string> words = new List<string>();
            string guess;
            string word;
            int attempts = 0;

            while(true){
                Console.Write("Enter a word: ");
                word = Console.ReadLine();
                if(string.IsNullOrEmpty(word)){
                    if(words.Count < 1){
                        continue;
                    }
                    break;
                }
                words.Add(word);
            }

            Random random = new Random();
            int number = random.Next(0, words.Count);
            word = words[number];

            while(attempts < 10){
                System.Console.Write("Guess a word: ");
                guess = Console.ReadLine();
                if(string.IsNullOrEmpty(guess)){
                    Console.WriteLine("You entered a non valid string!");
                    continue;
                }
                guesses.Add(guess);
                attempts++;

                if(word.Equals(guess)){
                    Console.WriteLine("That is correct!");
                    Console.WriteLine("Found answer in attempt: " + attempts);
                    break;
                }
                else{
                    Console.WriteLine("That is not correct!");
                    System.Console.WriteLine("Attempts made: " + attempts);
                }
            }

            Console.WriteLine("Guess history: ");
            foreach(string value in guesses){
                Console.WriteLine(value);
            }

            Console.WriteLine("Game over!");
        }
    }
}